baybe/symmetries.py

```diff
@@ -0,0 +1,439 @@
+"""Functionality for expressing symmetries of the modeling process."""
+
+from __future__ import annotations
+
+import gc
+from abc import ABC, abstractmethod
+from collections.abc import Iterable
+from itertools import combinations
+from typing import TYPE_CHECKING, Any, cast
+
+import numpy as np
+import pandas as pd
+from attrs import Converter, define, field
+from attrs.validators import deep_iterable, ge, instance_of, min_len
+from typing_extensions import override
+
+from baybe.constraints.conditions import Condition
+from baybe.serialization import SerialMixin
+from baybe.utils.augmentation import (
+    df_apply_dependency_augmentation,
+    df_apply_mirror_augmentation,
+    df_apply_permutation_augmentation,
+)
+from baybe.utils.conversion import normalize_str_sequence
+from baybe.utils.validation import validate_is_finite, validate_unique_values
+
+if TYPE_CHECKING:
+    from baybe.parameters.base import Parameter
+    from baybe.searchspace import SearchSpace
+
+
+@define(frozen=True)
+class Symmetry(SerialMixin, ABC):
+    """Abstract base class for symmetries.
+
+    Symmetry is a concept that can be used to configure the modelling process in the
+    presence of invariances.
+    """
+
+    use_data_augmentation: bool = field(
+        default=True, validator=instance_of(bool), kw_only=True
+    )
+    """Flag indicating whether data augmentation would be used with surrogates that
```

Comment:
Two questions/remarks:
* `would` sounds a bit weird, no? How about `is to be`
* Why the `that support this`? Isn't the beauty of the augmentation approach that it's surrogate-agnostic?